aptos-move/framework/aptos-names/sources/domains.move

```diff
@@ -157,6 +159,15 @@ module aptos_names::domains {
         set_name_address_internal(subdomain_name, domain_name, signer::address_of(sign));
     }
 
+    /// Generic function for registering domain with or without signature, depending on if unrestricted_mint_enabled is true
+    public entry fun register_domain_with_signature(sign: &signer, domain_name: String, num_years: u8, register_domain_signature: vector<u8>) acquires NameRegistryV1, RegisterNameEventsV1, SetNameAddressEventsV1 {
+        if (!unrestricted_mint_enabled()) {
```

Comment:
lets put this `unrestricted_mint_enabled` check into the regular `register_domain` entry method; this way we can enforce that they use `register_domain_with_signature` if required.
Then, within the `register_domain_with_signature` method we don't actually need to check if it's enabled or not: if they have a valid signature, we can trust the backend